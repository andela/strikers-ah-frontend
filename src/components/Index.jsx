/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import queryString from 'query-string';
import { connect } from 'react-redux';
import { getLoggedInUserProfile } from '../redux/actions/userAction';

/**
 * @author frank harerimana
 */
class Index extends Component {
  /**
   * @author frank
   * @returns {*} state
   */
  componentWillMount() {
    const user = localStorage.getItem('token');
    const url = this.props.location.search;
    const params = queryString.parse(url);
    if (params.token != null) {
      localStorage.setItem('token', params.token);
      this.props.getLoggedInUserProfile(params.token);
    } else if (user != null) {
      this.props.getLoggedInUserProfile(user);
    }
  }

  /**
   * @author frank harerimana
   * @returns {*} render
   */
  render() {
    if (this.props.userProfile.length !== 0) {
      const { user } = this.props.userProfile;
      console.log(user);
    }
    return (
      <div>
        <p>home page component</p>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  userProfile: state.userProfile,
});

export default connect(
  mapStateToProps,
  { getLoggedInUserProfile },
)(Index);
