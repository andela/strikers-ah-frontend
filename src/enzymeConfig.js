import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
