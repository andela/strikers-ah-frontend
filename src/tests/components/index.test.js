/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import '../../enzymeConfig';
// eslint-disable-next-line import/no-named-as-default
import Index from '../../components/Index';

const mockStore = configureStore();
const initialState = [{}];
const props = {
  location: {
    search: 'example.com',
  },
  componentWillMount: jest.fn(),
  socialLoginUser: {
    user: 'example',
  },
  getSocialUser: jest.fn(),
};
let store, wrapper;
describe('`Index.jsx`', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<Index store={store} {...props} />);
  });
  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  //   test('should call the componentWillMount', () => {
  //       expect(props.getSocialUser).toHaveBeenCalled();
  //   })
});
