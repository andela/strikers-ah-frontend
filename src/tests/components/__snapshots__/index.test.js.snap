// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Index.jsx\` should render 1`] = `ShallowWrapper {}`;
